src/app/components/button/button.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { buttonView, buttonType, iconPosition } from 'src/app/core/models/button.enum';
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss']
+})
+export class ButtonComponent {
+  typeOfButton = buttonType;
+  viewOfButton = buttonView;
+  iconOfButton= iconPosition;
+
+  @Input() buttonType: number = this.typeOfButton.fill;
+  @Input() buttonView: number = this.viewOfButton.small;
+  @Input() iconPlace?: number;
+  @Input() buttonLabel?: string = 'button';
+  @Input() buttonIcon?: string;
+
```

Comment:
Add input checker, for overlapping inputs, and throw an error in case of conflicts